### PR TITLE
Update 4.0.52

### PR DESCRIPTION
**Pull request by VJanKraemer**
* Add (default) option to vendor packages
* Make the rules file and build_deb.sh script cross platform for Ubuntu 20.04-24.04 (all current LTS releases)
* Update to SilKit 4.0.52 since this is the current UNRELEASED tag
* Minor fixes in the README

### CODE REVIEW

File: .github/actions/build_deb.sh [hunk 1]
```diff
@@ -35,51 +39,100 @@ if [ -n "${CI_RUN+1}" ] ; then
...
-if [ "$ret_val" != '0' ] ; then
-    echo "SILKIT-PKG: Could get SIL Kit sources at ${SILKIT_SOURCE_URL}:${SILKIT_REVISION}"
-    echo "Trying local directory at ${PWD}/libsilkit-${SILKIT_VERSION}"
-fi
+    # Since we want to allow building SIL KIT from VERSIONS different from the current CHANGELOG Version
+    # We need to distinguish how to checkout and reset the local git repo
+    if [ -n SILKIT_REVISION ] ; then
+        CLONE_VERSION="main"
+    else
+        CLONE_VERSION="sil-kit/v${SILKIT_VERSION}"
```
Comment (VDanielEdwards) [inline .github/actions/build_deb.sh:60]: The `if` above tests a variable named `SILKIT_REVISION`, but it uses `SILKIT_VERSION` - i this correct?

Comment (VDanielEdwards) [reply .github/actions/build_deb.sh:60]: Ah I see `SILKIT_REVISION` is the git ref.

**Review by VDanielEdwards — APPROVED** [review]
I'm certain that this script breaks in wonderful ways if a path contains a space, but apart from that it looks really nice! 👍 Thank you!

**Comment by VDanielEdwards** [comment]
> I'm certain that this script breaks in wonderful ways if a path contains a space, but apart from that it looks really nice! 👍 Thank you!

But really who does that on Linux 😺

File: .github/actions/build_deb.sh (hunk 1 above)
Comment (VJanKraemer) [reply .github/actions/build_deb.sh:60]: Yes if we specify the git ref, we just clone main and then reset to the correct ref!

**Pull request merged**